{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Working on Jsons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import json\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "output_type": "error",
     "ename": "SSLError",
     "evalue": "HTTPSConnectionPool(host='covid19tracking.narrativa.com', port=443): Max retries exceeded with url: /feed_en.json (Caused by SSLError(SSLError(1, '[SSL: WRONG_CIPHER_RETURNED] wrong cipher returned (_ssl.c:1108)')))",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    669\u001b[0m             \u001b[0;31m# Make the request on the httplib connection object.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 670\u001b[0;31m             httplib_response = self._make_request(\n\u001b[0m\u001b[1;32m    671\u001b[0m                 \u001b[0mconn\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    380\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 381\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_validate_conn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconn\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    382\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mSocketTimeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mBaseSSLError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_validate_conn\u001b[0;34m(self, conn)\u001b[0m\n\u001b[1;32m    975\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"sock\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# AppEngine might not have  `.sock`\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 976\u001b[0;31m             \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    977\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/connection.py\u001b[0m in \u001b[0;36mconnect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    360\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 361\u001b[0;31m         self.sock = ssl_wrap_socket(\n\u001b[0m\u001b[1;32m    362\u001b[0m             \u001b[0msock\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mconn\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/util/ssl_.py\u001b[0m in \u001b[0;36mssl_wrap_socket\u001b[0;34m(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password, ca_cert_data)\u001b[0m\n\u001b[1;32m    376\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mHAS_SNI\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mserver_hostname\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 377\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mcontext\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrap_socket\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msock\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mserver_hostname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mserver_hostname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    378\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ssl.py\u001b[0m in \u001b[0;36mwrap_socket\u001b[0;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)\u001b[0m\n\u001b[1;32m    499\u001b[0m         \u001b[0;31m# ctx._wrap_socket()\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 500\u001b[0;31m         return self.sslsocket_class._create(\n\u001b[0m\u001b[1;32m    501\u001b[0m             \u001b[0msock\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msock\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ssl.py\u001b[0m in \u001b[0;36m_create\u001b[0;34m(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)\u001b[0m\n\u001b[1;32m   1039\u001b[0m                         \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"do_handshake_on_connect should not be specified for non-blocking sockets\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1040\u001b[0;31m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdo_handshake\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1041\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mOSError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ssl.py\u001b[0m in \u001b[0;36mdo_handshake\u001b[0;34m(self, block)\u001b[0m\n\u001b[1;32m   1308\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msettimeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1309\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdo_handshake\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1310\u001b[0m         \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mSSLError\u001b[0m: [SSL: WRONG_CIPHER_RETURNED] wrong cipher returned (_ssl.c:1108)",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/adapters.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    438\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mchunked\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 439\u001b[0;31m                 resp = conn.urlopen(\n\u001b[0m\u001b[1;32m    440\u001b[0m                     \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    723\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 724\u001b[0;31m             retries = retries.increment(\n\u001b[0m\u001b[1;32m    725\u001b[0m                 \u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_pool\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_stacktrace\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexc_info\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/util/retry.py\u001b[0m in \u001b[0;36mincrement\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    438\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mnew_retry\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_exhausted\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 439\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mMaxRetryError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_pool\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merror\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mResponseError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcause\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    440\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mMaxRetryError\u001b[0m: HTTPSConnectionPool(host='covid19tracking.narrativa.com', port=443): Max retries exceeded with url: /feed_en.json (Caused by SSLError(SSLError(1, '[SSL: WRONG_CIPHER_RETURNED] wrong cipher returned (_ssl.c:1108)')))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-79-b70d2f8bfbcd>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'https://covid19tracking.narrativa.com/feed_en.json'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mjson_readed\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mjson\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloads\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mjson_readed\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/api.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     74\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     75\u001b[0m     \u001b[0mkwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetdefault\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'allow_redirects'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 76\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'get'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparams\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     77\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     78\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/api.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     59\u001b[0m     \u001b[0;31m# cases, and look like a memory leak in others.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     60\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0msessions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSession\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 61\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     62\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    528\u001b[0m         }\n\u001b[1;32m    529\u001b[0m         \u001b[0msend_kwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msettings\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 530\u001b[0;31m         \u001b[0mresp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprep\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0msend_kwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    531\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    532\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    641\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    642\u001b[0m         \u001b[0;31m# Send the request\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 643\u001b[0;31m         \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0madapter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    644\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    645\u001b[0m         \u001b[0;31m# Total elapsed time of the request (approximately)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/adapters.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    512\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreason\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_SSLError\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    513\u001b[0m                 \u001b[0;31m# This branch is for urllib3 v1.22 and later.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 514\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mSSLError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    515\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    516\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mConnectionError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mSSLError\u001b[0m: HTTPSConnectionPool(host='covid19tracking.narrativa.com', port=443): Max retries exceeded with url: /feed_en.json (Caused by SSLError(SSLError(1, '[SSL: WRONG_CIPHER_RETURNED] wrong cipher returned (_ssl.c:1108)')))"
     ]
    }
   ],
   "source": [
    "r = requests.get(url='https://covid19tracking.narrativa.com/feed_en.json')\n",
    "json_readed = json.loads(r.text)\n",
    "json_readed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "output_type": "error",
     "ename": "SSLError",
     "evalue": "HTTPSConnectionPool(host='covid19tracking.narrativa.com', port=443): Max retries exceeded with url: /feed_en.json (Caused by SSLError(SSLError(1, '[SSL: WRONG_CIPHER_RETURNED] wrong cipher returned (_ssl.c:1108)')))",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    669\u001b[0m             \u001b[0;31m# Make the request on the httplib connection object.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 670\u001b[0;31m             httplib_response = self._make_request(\n\u001b[0m\u001b[1;32m    671\u001b[0m                 \u001b[0mconn\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    380\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 381\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_validate_conn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconn\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    382\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mSocketTimeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mBaseSSLError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_validate_conn\u001b[0;34m(self, conn)\u001b[0m\n\u001b[1;32m    975\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"sock\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# AppEngine might not have  `.sock`\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 976\u001b[0;31m             \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    977\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/connection.py\u001b[0m in \u001b[0;36mconnect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    360\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 361\u001b[0;31m         self.sock = ssl_wrap_socket(\n\u001b[0m\u001b[1;32m    362\u001b[0m             \u001b[0msock\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mconn\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/util/ssl_.py\u001b[0m in \u001b[0;36mssl_wrap_socket\u001b[0;34m(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password, ca_cert_data)\u001b[0m\n\u001b[1;32m    376\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mHAS_SNI\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mserver_hostname\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 377\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mcontext\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrap_socket\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msock\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mserver_hostname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mserver_hostname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    378\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ssl.py\u001b[0m in \u001b[0;36mwrap_socket\u001b[0;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)\u001b[0m\n\u001b[1;32m    499\u001b[0m         \u001b[0;31m# ctx._wrap_socket()\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 500\u001b[0;31m         return self.sslsocket_class._create(\n\u001b[0m\u001b[1;32m    501\u001b[0m             \u001b[0msock\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msock\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ssl.py\u001b[0m in \u001b[0;36m_create\u001b[0;34m(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)\u001b[0m\n\u001b[1;32m   1039\u001b[0m                         \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"do_handshake_on_connect should not be specified for non-blocking sockets\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1040\u001b[0;31m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdo_handshake\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1041\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mOSError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ssl.py\u001b[0m in \u001b[0;36mdo_handshake\u001b[0;34m(self, block)\u001b[0m\n\u001b[1;32m   1308\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msettimeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1309\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdo_handshake\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1310\u001b[0m         \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mSSLError\u001b[0m: [SSL: WRONG_CIPHER_RETURNED] wrong cipher returned (_ssl.c:1108)",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/adapters.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    438\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mchunked\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 439\u001b[0;31m                 resp = conn.urlopen(\n\u001b[0m\u001b[1;32m    440\u001b[0m                     \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    723\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 724\u001b[0;31m             retries = retries.increment(\n\u001b[0m\u001b[1;32m    725\u001b[0m                 \u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_pool\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_stacktrace\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexc_info\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/urllib3/util/retry.py\u001b[0m in \u001b[0;36mincrement\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    438\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mnew_retry\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_exhausted\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 439\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mMaxRetryError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_pool\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merror\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mResponseError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcause\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    440\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mMaxRetryError\u001b[0m: HTTPSConnectionPool(host='covid19tracking.narrativa.com', port=443): Max retries exceeded with url: /feed_en.json (Caused by SSLError(SSLError(1, '[SSL: WRONG_CIPHER_RETURNED] wrong cipher returned (_ssl.c:1108)')))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-80-d0f1bb50e7e8>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"https://covid19tracking.narrativa.com/feed_en.json\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mjson_readed\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mjson_readed\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/api.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     74\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     75\u001b[0m     \u001b[0mkwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetdefault\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'allow_redirects'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 76\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'get'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparams\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     77\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     78\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/api.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     59\u001b[0m     \u001b[0;31m# cases, and look like a memory leak in others.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     60\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0msessions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSession\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 61\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     62\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    528\u001b[0m         }\n\u001b[1;32m    529\u001b[0m         \u001b[0msend_kwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msettings\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 530\u001b[0;31m         \u001b[0mresp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprep\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0msend_kwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    531\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    532\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    641\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    642\u001b[0m         \u001b[0;31m# Send the request\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 643\u001b[0;31m         \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0madapter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    644\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    645\u001b[0m         \u001b[0;31m# Total elapsed time of the request (approximately)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/adapters.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    512\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreason\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_SSLError\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    513\u001b[0m                 \u001b[0;31m# This branch is for urllib3 v1.22 and later.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 514\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mSSLError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    515\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    516\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mConnectionError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mSSLError\u001b[0m: HTTPSConnectionPool(host='covid19tracking.narrativa.com', port=443): Max retries exceeded with url: /feed_en.json (Caused by SSLError(SSLError(1, '[SSL: WRONG_CIPHER_RETURNED] wrong cipher returned (_ssl.c:1108)')))"
     ]
    }
   ],
   "source": [
    "r = requests.get(url = \"https://covid19tracking.narrativa.com/feed_en.json\")\n",
    "json_readed = r.json()\n",
    "\n",
    "json_readed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Save json in a local file called \"data.json\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('data.json', 'w+') as outfile:\n",
    "    json.dump(json_readed, outfile)\n",
    "\n",
    "#ceramos un fichero en local con w+(permisos de escritura y + que siginifica que, si no existe creamelo. Accedemos al metodo dump(volca los datos que hemos guardado en json_readed) que contiene la libreria json )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "tags": [
     "outputPrepend"
    ]
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": "41%2C132&region_confirmed_num=233%2C037&region_confirmed_today_num=344&region_deaths_today_num=48&region_deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.&top_paises2=Brasil&top_paises3=Rusia&top_paises1_num=1%2C681%2C418&top_paises2_num=391%2C222&top_paises3_num=370%2C680&top_comunidades1=Madrid&top_comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=13%3A10&client=default_web&region=world\", \"_id\": \"6658f6eb15b745e282dee2ccf6f0b3c0\", \"id\": \"2020-5-27-13_01-COVID-world-en_generic_COVID_EN_GENERIC\"}, {\"date\": \"2020-05-27T12:41:04+02:00\", \"metadata\": {\"picurl_square_spain\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&actives_num=54%2C721&recovered_num=150%2C376&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default\", \"picurl_square_spain_extended\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default&extended=true\", \"graphs\": {\"graph_picurl_china\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china.png\", \"graph_picurl_us\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us.png\", \"graph_picurl_deaths_evolution\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_evolution.png\", \"graph_picurl_world\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_world.png\", \"graph_picurl_china_confirmed\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_confirmed.png\", \"graph_picurl_us_confirmed\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_confirmed.png\", \"graph_picurl_spain_deaths\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_deaths.png\", \"graph_picurl_us_recovered\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_recovered.png\", \"graph_picurl_deaths_log_evolution\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_log_evolution.png\", \"graph_picurl_italy\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy.png\", \"graph_picurl_italy_recovered\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_recovered.png\", \"graph_picurl_us_deaths\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_deaths.png\", \"graph_picurl_italy_deaths\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_deaths.png\", \"graph_picurl_china_deaths\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_deaths.png\", \"graph_picurl_italy_confirmed\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_confirmed.png\", \"graph_picurl_spain_confirmed\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_confirmed.png\", \"graph_picurl_china_recovered\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_recovered.png\"}, \"picurl_square_world\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default\", \"picurl_16_9_spain\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&top_countries=EE.UU.%20%281%2C681%2C418%29%2C%20Brasil%20%28391%2C222%29%2C%20Rusia%20%28370%2C680%29&top_spain=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&region=spain&aspect_ratio=16_9&client=default\", \"composition_type\": \"world\", \"country_selected\": null, \"picurl_square_world_extended\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default&extended=true\", \"state_selected\": null}, \"author\": \"Narrativa\", \"_rev\": \"1-2f39e9c455969a32a34a2b654f77c1d0\", \"language\": \"en_generic\", \"title\": \"Coronavirus outbreak latest: <strong>5,557,329</strong> people infected worldwide\", \"content\": \"<p><small>Narrativa | Wednesday, May 27, 2020 </small></p><p>The number of COVID-19 cases worldwide is now at <strong>5,557,329</strong>, of which <strong>2,298,685</strong></strong> have already recovered, according to the data <a href=\\\"https://systems.jhu.edu/research/public-health/ncov/\\\" target=\\\"_blank\\\">provided by Johns Hopkins University</a>. <b>The US</b> has the most cases with <strong>1,681,418</strong>, followed by <b> Brazil</b>, <b> Russia</b>, and <b>The United Kingdom</b>, who have confirmed <strong>391,222, 370,680, and 266,599</strong> cases respectively.</p><h4>Status in the world</h4><p><table width=\\\"100%\\\"><thead><tr><th><strong>Currently Infected</strong></th><th colspan=\\\"2\\\"><strong>Infected</strong></th><th colspan=\\\"2\\\"><strong>Deaths</strong></th><th colspan=\\\"2\\\"><strong>Recovered</strong></th></tr><tr><th>Total</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th></tr></thead><tbody><tr><td>1. The US: 1,197,587</td><td>1. The US: 1,681,418</td><td>1. Russia: 8,338</td><td>1. The US: 98,929</td><td>1. Russia: 161</td><td>1. The US: 384,902</strong></td><td>1. Russia: 11,079</strong></td></tr><tr><td>2. The United Kingdom: 228,307</td><td>2. Brazil: 391,222</td><td>2. Bangladesh: 1,541</td><td>2. The United Kingdom: 37,130</td><td>2. France: 83</td><td>2. Germany: 162,690</strong></td><td>2. Italy: 2,677</strong></td></tr><tr><td>3. Russia: 224,504</td><td>3. Russia: 370,680</td><td>3. Pakistan: 1,446</td><td>3. Italy: 32,955</td><td>3. The United Kingdom: 82</td><td>3. Brazil: 158,593</strong></td><td>3. Mexico: 2,256</strong></td></tr></tbody></table></p><p>Moreover, <strong>22,010</strong> new positive cases have been added in the last 24 hours, demonstrating a <strong>0.4</strong>% growth from the previous day. <strong> Russia</strong> is the country with the most new cases, having now confirmed <strong>8,338</strong>, followed by <b> Bangladesh</b> with <strong>1,541</strong>. <strong> Pakistan</strong> and <strong>The United Kingdom</strong> are not far behind, with <strong>1,372</strong> and <strong>1,372</strong> new patients respectively.</p><p> </p><p>So far, <strong> 341,118 people</strong> have died from the coronavirus worldwide. <strong>The US</strong> has the most victims so far with <strong>98,929</strong> deceased, followed by <strong>The United Kingdom, Italy, and Spain</strong> with <strong>37,130, 32,955, and 27,940</strong> deaths respectively.</p><p>The first evidences about this new strain of the virus appeared in <strong>China</strong>, in the city of <strong>Wuhan</strong>, on <strong>31 December 2019</strong>. Since then, the infection has spread across Asia and the world. At the moment, cases in China have grown to <strong>1</strong> new cases since Tuesday. Overall, <strong>105</strong> people are infected with the virus.</p><hr><p><strong>Note</strong>: This article has been <strong><u>generated automatically by <a href=\\\"https://www.narrativa.com\\\" target=\\\"_blank\\\">Narrativa</a></u></strong> from the data by <a href=\\\"https://www.mscbs.gob.es/profesionales/saludPublica/ccayes/alertasActual/nCov-China/situacionActual.htm\\\" target=\\\"_blank\\\">Ministerio de Sanidad</a>, <a href=\\\"https://github.com/pcm-dpc/COVID-19\\\" target=\\\"_blank\\\">Dipartimento della Protezione Civile de Italia</a>, <a href=\\\"https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/Gesamt.html\\\" target=\\\"_blank\\\">Robert Koch Institute</a> and <a href=\\\"https://systems.jhu.edu/research/public-health/ncov/\\\" target=\\\"_blank\\\">Johns Hopkins University</a>.</p>\", \"picurl_world\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.%3A%201%2C681%2C418&top_paises2=Brasil%3A%20391%2C222&top_paises3=Rusia%3A%20370%2C680&client=default&region=world\", \"picurl\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_comunidades=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&client=default&region=spain\", \"metatitle\": \"\", \"subtitle\": \"\", \"picurl_banner_web\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&region_confirmed_num=233%2C037&region_confirmed_today_num=344&region_deaths_today_num=48&region_deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.&top_paises2=Brasil&top_paises3=Rusia&top_paises1_num=1%2C681%2C418&top_paises2_num=391%2C222&top_paises3_num=370%2C680&top_comunidades1=Madrid&top_comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=12%3A36&client=default_web&region=world\", \"_id\": \"6658f6eb15b745e282dee2ccf6def733\", \"id\": \"2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC\"}, {\"date\": \"2020-05-27T11:06:53+02:00\", \"metadata\": {\"picurl_square_spain\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&actives_num=54%2C721&recovered_num=150%2C376&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default\", \"picurl_square_spain_extended\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default&extended=true\", \"graphs\": {\"graph_picurl_china\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china.png\", \"graph_picurl_us\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us.png\", \"graph_picurl_deaths_evolution\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_evolution.png\", \"graph_picurl_world\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_world.png\", \"graph_picurl_china_confirmed\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_confirmed.png\", \"graph_picurl_us_confirmed\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_confirmed.png\", \"graph_picurl_spain_deaths\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_deaths.png\", \"graph_picurl_us_recovered\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_recovered.png\", \"graph_picurl_deaths_log_evolution\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_log_evolution.png\", \"graph_picurl_italy\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy.png\", \"graph_picurl_italy_recovered\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_recovered.png\", \"graph_picurl_us_deaths\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_deaths.png\", \"graph_picurl_italy_deaths\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_deaths.png\", \"graph_picurl_china_deaths\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_deaths.png\", \"graph_picurl_italy_confirmed\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_confirmed.png\", \"graph_picurl_spain_confirmed\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_confirmed.png\", \"graph_picurl_china_recovered\": \"https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_recovered.png\"}, \"picurl_square_world\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default\", \"picurl_16_9_spain\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&top_countries=EE.UU.%20%281%2C681%2C418%29%2C%20Brasil%20%28391%2C222%29%2C%20Rusia%20%28370%2C680%29&top_spain=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&region=spain&aspect_ratio=16_9&client=default\", \"composition_type\": \"world\", \"country_selected\": null, \"picurl_square_world_extended\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default&extended=true\", \"state_selected\": null}, \"author\": \"Narrativa\", \"_rev\": \"1-e9ca35acd2d130b04289e3333083ca14\", \"language\": \"en_generic\", \"title\": \"Coronavirus outbreak latest: <strong>341,008</strong> deaths confirmed around the world\", \"content\": \"<p><small>Narrativa | Wednesday, May 27, 2020 </small></p><p>The number of COVID-19 cases worldwide is now at <strong>5,554,865</strong> Amongst this number, <strong>2,297,943</strong></strong> have already recovered, according to the data <a href=\\\"https://systems.jhu.edu/research/public-health/ncov/\\\" target=\\\"_blank\\\">provided by Johns Hopkins University</a>. <b>The US</b> tops the list with <strong>1,681,418</strong> recorded cases, followed by <b> Brazil</b>, <b> Russia</b>, and <b>The United Kingdom</b> with <strong>391,222, 370,680, and 266,599</strong> cases respectively.</p><h4>COVID status across most infected countries</h4><p><table width=\\\"100%\\\"><thead><tr><th><strong>Active Cases</strong></th><th colspan=\\\"2\\\"><strong>Infected</strong></th><th colspan=\\\"2\\\"><strong>Deaths</strong></th><th colspan=\\\"2\\\"><strong>Recovered</strong></th></tr><tr><th>Total</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th></tr></thead><tbody><tr><td>1. The US: 1,197,587</td><td>1. The US: 1,681,418</td><td>1. Russia: 8,338</td><td>1. The US: 98,929</td><td>1. Russia: 161</td><td>1. The US: 384,902</strong></td><td>1. Russia: 11,079</strong></td></tr><tr><td>2. The United Kingdom: 228,307</td><td>2. Brazil: 391,222</td><td>2. Pakistan: 1,446</td><td>2. The United Kingdom: 37,130</td><td>2. France: 83</td><td>2. Germany: 162,690</strong></td><td>2. Italy: 2,677</strong></td></tr><tr><td>3. Russia: 224,504</td><td>3. Russia: 370,680</td><td>3. The United Kingdom: 1,372</td><td>3. Italy: 32,955</td><td>3. The United Kingdom: 82</td><td>3. Brazil: 158,593</strong></td><td>3. Mexico: 2,256</strong></td></tr></tbody></table></p><p>Moreover, the last 24 hours have seen <strong>19,546</strong> new cases added in the last 24 hours, an increase of <strong>0.35</strong>% from the previous day. <strong> Russia</strong> is the country with the most new cases, having now confirmed <strong>8,338</strong>, followed by <b> Pakistan</b> with <strong>1,446</strong>. <strong>The United Kingdom</strong> and <strong>The US</strong> are not far behind, with <strong>1,117</strong> and <strong>1,117</strong> new patients respectively.</p><p> </p><p>So far, <strong> 341,008 people</strong> have died from the coronavirus worldwide. <strong>The US</strong> is the country with the highest count so far with <strong>98,929</strong> deceased, followed by <strong>The United Kingdom, Italy, and Spain</strong> with <strong>37,130, 32,955, and 27,940</strong> deaths respectively.</p><p>The first news of this new strain of the virus appeared in <strong>China</strong>, in the city of <strong>Wuhan</strong>, on <strong>31 December 2019</strong>. Since that date, the number of infections have multiplied across the continent and the world. At the moment, cases in China have increased to <strong>1</strong> new cases since yesterday. Overall, <strong>105</strong> people are infected with coronavirus.</p><hr><p><strong>Note</strong>: This article has been <strong><u>generated automatically by <a href=\\\"https://www.narrativa.com\\\" target=\\\"_blank\\\">Narrativa</a></u></strong> from the data by <a href=\\\"https://www.mscbs.gob.es/profesionales/saludPublica/ccayes/alertasActual/nCov-China/situacionActual.htm\\\" target=\\\"_blank\\\">Ministerio de Sanidad</a>, <a href=\\\"https://github.com/pcm-dpc/COVID-19\\\" target=\\\"_blank\\\">Dipartimento della Protezione Civile de Italia</a>, <a href=\\\"https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/Gesamt.html\\\" target=\\\"_blank\\\">Robert Koch Institute</a> and <a href=\\\"https://systems.jhu.edu/research/public-health/ncov/\\\" target=\\\"_blank\\\">Johns Hopkins University</a>.</p>\", \"picurl_world\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.%3A%201%2C681%2C418&top_paises2=Brasil%3A%20391%2C222&top_paises3=Rusia%3A%20370%2C680&client=default&region=world\", \"picurl\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_comunidades=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&client=default&region=spain\", \"metatitle\": \"\", \"subtitle\": \"\", \"picurl_banner_web\": \"https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&region_confirmed_num=233%2C037&region_confirmed_today_num=344&region_deaths_today_num=48&region_deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.&top_paises2=Brasil&top_paises3=Rusia&top_paises1_num=1%2C681%2C418&top_paises2_num=391%2C222&top_paises3_num=370%2C680&top_comunidades1=Madrid&top_comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=11%3A03&client=default_web&region=world\", \"_id\": \"6658f6eb15b745e282dee2ccf67bb59d\", \"id\": \"2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC\"}]}}\n\n"
    }
   ],
   "source": [
    "with open('data.json') as f:\n",
    "    line = f.readline()\n",
    "    if line:\n",
    "        print(line)\n",
    "        line = f.readline()\n",
    "print(line)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Save with indent"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=13%3A10&client=default_web&region=world',\n    '_id': '6658f6eb15b745e282dee2ccf6f0b3c0',\n    'id': '2020-5-27-13_01-COVID-world-en_generic_COVID_EN_GENERIC'},\n   {'date': '2020-05-27T12:41:04+02:00',\n    'metadata': {'picurl_square_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&actives_num=54%2C721&recovered_num=150%2C376&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default',\n     'picurl_square_spain_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default&extended=true',\n     'graphs': {'graph_picurl_china': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china.png',\n      'graph_picurl_us': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us.png',\n      'graph_picurl_deaths_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_evolution.png',\n      'graph_picurl_world': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_world.png',\n      'graph_picurl_china_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_confirmed.png',\n      'graph_picurl_us_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_confirmed.png',\n      'graph_picurl_spain_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_deaths.png',\n      'graph_picurl_us_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_recovered.png',\n      'graph_picurl_deaths_log_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_log_evolution.png',\n      'graph_picurl_italy': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy.png',\n      'graph_picurl_italy_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_recovered.png',\n      'graph_picurl_us_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_deaths.png',\n      'graph_picurl_italy_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_deaths.png',\n      'graph_picurl_china_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_deaths.png',\n      'graph_picurl_italy_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_confirmed.png',\n      'graph_picurl_spain_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_confirmed.png',\n      'graph_picurl_china_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_recovered.png'},\n     'picurl_square_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default',\n     'picurl_16_9_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&top_countries=EE.UU.%20%281%2C681%2C418%29%2C%20Brasil%20%28391%2C222%29%2C%20Rusia%20%28370%2C680%29&top_spain=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&region=spain&aspect_ratio=16_9&client=default',\n     'composition_type': 'world',\n     'country_selected': None,\n     'picurl_square_world_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default&extended=true',\n     'state_selected': None},\n    'author': 'Narrativa',\n    '_rev': '1-2f39e9c455969a32a34a2b654f77c1d0',\n    'language': 'en_generic',\n    'title': 'Coronavirus outbreak latest: <strong>5,557,329</strong> people infected worldwide',\n    'content': '<p><small>Narrativa | Wednesday, May 27, 2020 </small></p><p>The number of COVID-19 cases worldwide is now at <strong>5,557,329</strong>, of which <strong>2,298,685</strong></strong> have already recovered, according to the data <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">provided by Johns Hopkins University</a>. <b>The US</b> has the most cases with <strong>1,681,418</strong>, followed by <b> Brazil</b>, <b> Russia</b>, and <b>The United Kingdom</b>, who have confirmed <strong>391,222, 370,680, and 266,599</strong> cases respectively.</p><h4>Status in the world</h4><p><table width=\"100%\"><thead><tr><th><strong>Currently Infected</strong></th><th colspan=\"2\"><strong>Infected</strong></th><th colspan=\"2\"><strong>Deaths</strong></th><th colspan=\"2\"><strong>Recovered</strong></th></tr><tr><th>Total</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th></tr></thead><tbody><tr><td>1. The US: 1,197,587</td><td>1. The US: 1,681,418</td><td>1. Russia: 8,338</td><td>1. The US: 98,929</td><td>1. Russia: 161</td><td>1. The US: 384,902</strong></td><td>1. Russia: 11,079</strong></td></tr><tr><td>2. The United Kingdom: 228,307</td><td>2. Brazil: 391,222</td><td>2. Bangladesh: 1,541</td><td>2. The United Kingdom: 37,130</td><td>2. France: 83</td><td>2. Germany: 162,690</strong></td><td>2. Italy: 2,677</strong></td></tr><tr><td>3. Russia: 224,504</td><td>3. Russia: 370,680</td><td>3. Pakistan: 1,446</td><td>3. Italy: 32,955</td><td>3. The United Kingdom: 82</td><td>3. Brazil: 158,593</strong></td><td>3. Mexico: 2,256</strong></td></tr></tbody></table></p><p>Moreover, <strong>22,010</strong> new positive cases have been added in the last 24 hours, demonstrating a <strong>0.4</strong>% growth from the previous day. <strong> Russia</strong> is the country with the most new cases, having now confirmed <strong>8,338</strong>, followed by <b> Bangladesh</b> with <strong>1,541</strong>. <strong> Pakistan</strong> and <strong>The United Kingdom</strong> are not far behind, with <strong>1,372</strong> and <strong>1,372</strong> new patients respectively.</p><p> </p><p>So far, <strong> 341,118 people</strong> have died from the coronavirus worldwide. <strong>The US</strong> has the most victims so far with <strong>98,929</strong> deceased, followed by <strong>The United Kingdom, Italy, and Spain</strong> with <strong>37,130, 32,955, and 27,940</strong> deaths respectively.</p><p>The first evidences about this new strain of the virus appeared in <strong>China</strong>, in the city of <strong>Wuhan</strong>, on <strong>31 December 2019</strong>. Since then, the infection has spread across Asia and the world. At the moment, cases in China have grown to <strong>1</strong> new cases since Tuesday. Overall, <strong>105</strong> people are infected with the virus.</p><hr><p><strong>Note</strong>: This article has been <strong><u>generated automatically by <a href=\"https://www.narrativa.com\" target=\"_blank\">Narrativa</a></u></strong> from the data by <a href=\"https://www.mscbs.gob.es/profesionales/saludPublica/ccayes/alertasActual/nCov-China/situacionActual.htm\" target=\"_blank\">Ministerio de Sanidad</a>, <a href=\"https://github.com/pcm-dpc/COVID-19\" target=\"_blank\">Dipartimento della Protezione Civile de Italia</a>, <a href=\"https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/Gesamt.html\" target=\"_blank\">Robert Koch Institute</a> and <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">Johns Hopkins University</a>.</p>',\n    'picurl_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.%3A%201%2C681%2C418&top_paises2=Brasil%3A%20391%2C222&top_paises3=Rusia%3A%20370%2C680&client=default&region=world',\n    'picurl': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_comunidades=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&client=default&region=spain',\n    'metatitle': '',\n    'subtitle': '',\n    'picurl_banner_web': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&region_confirmed_num=233%2C037&region_confirmed_today_num=344&region_deaths_today_num=48&region_deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.&top_paises2=Brasil&top_paises3=Rusia&top_paises1_num=1%2C681%2C418&top_paises2_num=391%2C222&top_paises3_num=370%2C680&top_comunidades1=Madrid&top_comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=12%3A36&client=default_web&region=world',\n    '_id': '6658f6eb15b745e282dee2ccf6def733',\n    'id': '2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC'},\n   {'date': '2020-05-27T11:06:53+02:00',\n    'metadata': {'picurl_square_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&actives_num=54%2C721&recovered_num=150%2C376&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default',\n     'picurl_square_spain_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default&extended=true',\n     'graphs': {'graph_picurl_china': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china.png',\n      'graph_picurl_us': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us.png',\n      'graph_picurl_deaths_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_evolution.png',\n      'graph_picurl_world': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_world.png',\n      'graph_picurl_china_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_confirmed.png',\n      'graph_picurl_us_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_confirmed.png',\n      'graph_picurl_spain_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_deaths.png',\n      'graph_picurl_us_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_recovered.png',\n      'graph_picurl_deaths_log_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_log_evolution.png',\n      'graph_picurl_italy': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy.png',\n      'graph_picurl_italy_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_recovered.png',\n      'graph_picurl_us_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_deaths.png',\n      'graph_picurl_italy_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_deaths.png',\n      'graph_picurl_china_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_deaths.png',\n      'graph_picurl_italy_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_confirmed.png',\n      'graph_picurl_spain_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_confirmed.png',\n      'graph_picurl_china_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_recovered.png'},\n     'picurl_square_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default',\n     'picurl_16_9_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&top_countries=EE.UU.%20%281%2C681%2C418%29%2C%20Brasil%20%28391%2C222%29%2C%20Rusia%20%28370%2C680%29&top_spain=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&region=spain&aspect_ratio=16_9&client=default',\n     'composition_type': 'world',\n     'country_selected': None,\n     'picurl_square_world_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default&extended=true',\n     'state_selected': None},\n    'author': 'Narrativa',\n    '_rev': '1-e9ca35acd2d130b04289e3333083ca14',\n    'language': 'en_generic',\n    'title': 'Coronavirus outbreak latest: <strong>341,008</strong> deaths confirmed around the world',\n    'content': '<p><small>Narrativa | Wednesday, May 27, 2020 </small></p><p>The number of COVID-19 cases worldwide is now at <strong>5,554,865</strong> Amongst this number, <strong>2,297,943</strong></strong> have already recovered, according to the data <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">provided by Johns Hopkins University</a>. <b>The US</b> tops the list with <strong>1,681,418</strong> recorded cases, followed by <b> Brazil</b>, <b> Russia</b>, and <b>The United Kingdom</b> with <strong>391,222, 370,680, and 266,599</strong> cases respectively.</p><h4>COVID status across most infected countries</h4><p><table width=\"100%\"><thead><tr><th><strong>Active Cases</strong></th><th colspan=\"2\"><strong>Infected</strong></th><th colspan=\"2\"><strong>Deaths</strong></th><th colspan=\"2\"><strong>Recovered</strong></th></tr><tr><th>Total</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th></tr></thead><tbody><tr><td>1. The US: 1,197,587</td><td>1. The US: 1,681,418</td><td>1. Russia: 8,338</td><td>1. The US: 98,929</td><td>1. Russia: 161</td><td>1. The US: 384,902</strong></td><td>1. Russia: 11,079</strong></td></tr><tr><td>2. The United Kingdom: 228,307</td><td>2. Brazil: 391,222</td><td>2. Pakistan: 1,446</td><td>2. The United Kingdom: 37,130</td><td>2. France: 83</td><td>2. Germany: 162,690</strong></td><td>2. Italy: 2,677</strong></td></tr><tr><td>3. Russia: 224,504</td><td>3. Russia: 370,680</td><td>3. The United Kingdom: 1,372</td><td>3. Italy: 32,955</td><td>3. The United Kingdom: 82</td><td>3. Brazil: 158,593</strong></td><td>3. Mexico: 2,256</strong></td></tr></tbody></table></p><p>Moreover, the last 24 hours have seen <strong>19,546</strong> new cases added in the last 24 hours, an increase of <strong>0.35</strong>% from the previous day. <strong> Russia</strong> is the country with the most new cases, having now confirmed <strong>8,338</strong>, followed by <b> Pakistan</b> with <strong>1,446</strong>. <strong>The United Kingdom</strong> and <strong>The US</strong> are not far behind, with <strong>1,117</strong> and <strong>1,117</strong> new patients respectively.</p><p> </p><p>So far, <strong> 341,008 people</strong> have died from the coronavirus worldwide. <strong>The US</strong> is the country with the highest count so far with <strong>98,929</strong> deceased, followed by <strong>The United Kingdom, Italy, and Spain</strong> with <strong>37,130, 32,955, and 27,940</strong> deaths respectively.</p><p>The first news of this new strain of the virus appeared in <strong>China</strong>, in the city of <strong>Wuhan</strong>, on <strong>31 December 2019</strong>. Since that date, the number of infections have multiplied across the continent and the world. At the moment, cases in China have increased to <strong>1</strong> new cases since yesterday. Overall, <strong>105</strong> people are infected with coronavirus.</p><hr><p><strong>Note</strong>: This article has been <strong><u>generated automatically by <a href=\"https://www.narrativa.com\" target=\"_blank\">Narrativa</a></u></strong> from the data by <a href=\"https://www.mscbs.gob.es/profesionales/saludPublica/ccayes/alertasActual/nCov-China/situacionActual.htm\" target=\"_blank\">Ministerio de Sanidad</a>, <a href=\"https://github.com/pcm-dpc/COVID-19\" target=\"_blank\">Dipartimento della Protezione Civile de Italia</a>, <a href=\"https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/Gesamt.html\" target=\"_blank\">Robert Koch Institute</a> and <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">Johns Hopkins University</a>.</p>',\n    'picurl_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.%3A%201%2C681%2C418&top_paises2=Brasil%3A%20391%2C222&top_paises3=Rusia%3A%20370%2C680&client=default&region=world',\n    'picurl': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_comunidades=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&client=default&region=spain',\n    'metatitle': '',\n    'subtitle': '',\n    'picurl_banner_web': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&region_confirmed_num=233%2C037&region_confirmed_today_num=344&region_deaths_today_num=48&region_deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.&top_paises2=Brasil&top_paises3=Rusia&top_paises1_num=1%2C681%2C418&top_paises2_num=391%2C222&top_paises3_num=370%2C680&top_comunidades1=Madrid&top_comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=11%3A03&client=default_web&region=world',\n    '_id': '6658f6eb15b745e282dee2ccf67bb59d',\n    'id': '2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC'}]}}"
     },
     "metadata": {},
     "execution_count": 83
    }
   ],
   "source": [
    "with open('data_indented.json', 'w+') as outfile:\n",
    "    json.dump(json_readed, outfile, indent=4)\n",
    "json_readed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read local json\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "output_type": "error",
     "ename": "ValueError",
     "evalue": "I/O operation on closed file.",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-84-5d965f39e3c6>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"data_indented.json\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"r+\"\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0moutflile\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m     \u001b[0mjson_readed\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mjson\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mload\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moutfile\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mjson_readed\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/__init__.py\u001b[0m in \u001b[0;36mload\u001b[0;34m(fp, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)\u001b[0m\n\u001b[1;32m    291\u001b[0m     \u001b[0mkwarg\u001b[0m\u001b[0;34m;\u001b[0m \u001b[0motherwise\u001b[0m\u001b[0;31m \u001b[0m\u001b[0;31m`\u001b[0m\u001b[0;31m`\u001b[0m\u001b[0mJSONDecoder\u001b[0m\u001b[0;31m`\u001b[0m\u001b[0;31m`\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mused\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    292\u001b[0m     \"\"\"\n\u001b[0;32m--> 293\u001b[0;31m     return loads(fp.read(),\n\u001b[0m\u001b[1;32m    294\u001b[0m         \u001b[0mcls\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcls\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mobject_hook\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mobject_hook\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    295\u001b[0m         \u001b[0mparse_float\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparse_float\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparse_int\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparse_int\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: I/O operation on closed file."
     ]
    }
   ],
   "source": [
    "with open(\"data_indented.json\", \"r+\") as outflile:\n",
    "    json_readed = json.load(outfile)\n",
    "json_readed\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=13%3A10&client=default_web&region=world',\n    '_id': '6658f6eb15b745e282dee2ccf6f0b3c0',\n    'id': '2020-5-27-13_01-COVID-world-en_generic_COVID_EN_GENERIC'},\n   {'date': '2020-05-27T12:41:04+02:00',\n    'metadata': {'picurl_square_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&actives_num=54%2C721&recovered_num=150%2C376&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default',\n     'picurl_square_spain_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default&extended=true',\n     'graphs': {'graph_picurl_china': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china.png',\n      'graph_picurl_us': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us.png',\n      'graph_picurl_deaths_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_evolution.png',\n      'graph_picurl_world': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_world.png',\n      'graph_picurl_china_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_confirmed.png',\n      'graph_picurl_us_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_confirmed.png',\n      'graph_picurl_spain_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_deaths.png',\n      'graph_picurl_us_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_recovered.png',\n      'graph_picurl_deaths_log_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_log_evolution.png',\n      'graph_picurl_italy': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy.png',\n      'graph_picurl_italy_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_recovered.png',\n      'graph_picurl_us_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_deaths.png',\n      'graph_picurl_italy_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_deaths.png',\n      'graph_picurl_china_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_deaths.png',\n      'graph_picurl_italy_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_confirmed.png',\n      'graph_picurl_spain_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_confirmed.png',\n      'graph_picurl_china_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_recovered.png'},\n     'picurl_square_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default',\n     'picurl_16_9_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&top_countries=EE.UU.%20%281%2C681%2C418%29%2C%20Brasil%20%28391%2C222%29%2C%20Rusia%20%28370%2C680%29&top_spain=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&region=spain&aspect_ratio=16_9&client=default',\n     'composition_type': 'world',\n     'country_selected': None,\n     'picurl_square_world_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default&extended=true',\n     'state_selected': None},\n    'author': 'Narrativa',\n    '_rev': '1-2f39e9c455969a32a34a2b654f77c1d0',\n    'language': 'en_generic',\n    'title': 'Coronavirus outbreak latest: <strong>5,557,329</strong> people infected worldwide',\n    'content': '<p><small>Narrativa | Wednesday, May 27, 2020 </small></p><p>The number of COVID-19 cases worldwide is now at <strong>5,557,329</strong>, of which <strong>2,298,685</strong></strong> have already recovered, according to the data <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">provided by Johns Hopkins University</a>. <b>The US</b> has the most cases with <strong>1,681,418</strong>, followed by <b> Brazil</b>, <b> Russia</b>, and <b>The United Kingdom</b>, who have confirmed <strong>391,222, 370,680, and 266,599</strong> cases respectively.</p><h4>Status in the world</h4><p><table width=\"100%\"><thead><tr><th><strong>Currently Infected</strong></th><th colspan=\"2\"><strong>Infected</strong></th><th colspan=\"2\"><strong>Deaths</strong></th><th colspan=\"2\"><strong>Recovered</strong></th></tr><tr><th>Total</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th></tr></thead><tbody><tr><td>1. The US: 1,197,587</td><td>1. The US: 1,681,418</td><td>1. Russia: 8,338</td><td>1. The US: 98,929</td><td>1. Russia: 161</td><td>1. The US: 384,902</strong></td><td>1. Russia: 11,079</strong></td></tr><tr><td>2. The United Kingdom: 228,307</td><td>2. Brazil: 391,222</td><td>2. Bangladesh: 1,541</td><td>2. The United Kingdom: 37,130</td><td>2. France: 83</td><td>2. Germany: 162,690</strong></td><td>2. Italy: 2,677</strong></td></tr><tr><td>3. Russia: 224,504</td><td>3. Russia: 370,680</td><td>3. Pakistan: 1,446</td><td>3. Italy: 32,955</td><td>3. The United Kingdom: 82</td><td>3. Brazil: 158,593</strong></td><td>3. Mexico: 2,256</strong></td></tr></tbody></table></p><p>Moreover, <strong>22,010</strong> new positive cases have been added in the last 24 hours, demonstrating a <strong>0.4</strong>% growth from the previous day. <strong> Russia</strong> is the country with the most new cases, having now confirmed <strong>8,338</strong>, followed by <b> Bangladesh</b> with <strong>1,541</strong>. <strong> Pakistan</strong> and <strong>The United Kingdom</strong> are not far behind, with <strong>1,372</strong> and <strong>1,372</strong> new patients respectively.</p><p> </p><p>So far, <strong> 341,118 people</strong> have died from the coronavirus worldwide. <strong>The US</strong> has the most victims so far with <strong>98,929</strong> deceased, followed by <strong>The United Kingdom, Italy, and Spain</strong> with <strong>37,130, 32,955, and 27,940</strong> deaths respectively.</p><p>The first evidences about this new strain of the virus appeared in <strong>China</strong>, in the city of <strong>Wuhan</strong>, on <strong>31 December 2019</strong>. Since then, the infection has spread across Asia and the world. At the moment, cases in China have grown to <strong>1</strong> new cases since Tuesday. Overall, <strong>105</strong> people are infected with the virus.</p><hr><p><strong>Note</strong>: This article has been <strong><u>generated automatically by <a href=\"https://www.narrativa.com\" target=\"_blank\">Narrativa</a></u></strong> from the data by <a href=\"https://www.mscbs.gob.es/profesionales/saludPublica/ccayes/alertasActual/nCov-China/situacionActual.htm\" target=\"_blank\">Ministerio de Sanidad</a>, <a href=\"https://github.com/pcm-dpc/COVID-19\" target=\"_blank\">Dipartimento della Protezione Civile de Italia</a>, <a href=\"https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/Gesamt.html\" target=\"_blank\">Robert Koch Institute</a> and <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">Johns Hopkins University</a>.</p>',\n    'picurl_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.%3A%201%2C681%2C418&top_paises2=Brasil%3A%20391%2C222&top_paises3=Rusia%3A%20370%2C680&client=default&region=world',\n    'picurl': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_comunidades=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&client=default&region=spain',\n    'metatitle': '',\n    'subtitle': '',\n    'picurl_banner_web': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&region_confirmed_num=233%2C037&region_confirmed_today_num=344&region_deaths_today_num=48&region_deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.&top_paises2=Brasil&top_paises3=Rusia&top_paises1_num=1%2C681%2C418&top_paises2_num=391%2C222&top_paises3_num=370%2C680&top_comunidades1=Madrid&top_comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=12%3A36&client=default_web&region=world',\n    '_id': '6658f6eb15b745e282dee2ccf6def733',\n    'id': '2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC'},\n   {'date': '2020-05-27T11:06:53+02:00',\n    'metadata': {'picurl_square_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&actives_num=54%2C721&recovered_num=150%2C376&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default',\n     'picurl_square_spain_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default&extended=true',\n     'graphs': {'graph_picurl_china': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china.png',\n      'graph_picurl_us': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us.png',\n      'graph_picurl_deaths_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_evolution.png',\n      'graph_picurl_world': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_world.png',\n      'graph_picurl_china_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_confirmed.png',\n      'graph_picurl_us_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_confirmed.png',\n      'graph_picurl_spain_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_deaths.png',\n      'graph_picurl_us_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_recovered.png',\n      'graph_picurl_deaths_log_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_log_evolution.png',\n      'graph_picurl_italy': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy.png',\n      'graph_picurl_italy_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_recovered.png',\n      'graph_picurl_us_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_deaths.png',\n      'graph_picurl_italy_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_deaths.png',\n      'graph_picurl_china_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_deaths.png',\n      'graph_picurl_italy_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_confirmed.png',\n      'graph_picurl_spain_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_confirmed.png',\n      'graph_picurl_china_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_recovered.png'},\n     'picurl_square_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default',\n     'picurl_16_9_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&top_countries=EE.UU.%20%281%2C681%2C418%29%2C%20Brasil%20%28391%2C222%29%2C%20Rusia%20%28370%2C680%29&top_spain=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&region=spain&aspect_ratio=16_9&client=default',\n     'composition_type': 'world',\n     'country_selected': None,\n     'picurl_square_world_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default&extended=true',\n     'state_selected': None},\n    'author': 'Narrativa',\n    '_rev': '1-e9ca35acd2d130b04289e3333083ca14',\n    'language': 'en_generic',\n    'title': 'Coronavirus outbreak latest: <strong>341,008</strong> deaths confirmed around the world',\n    'content': '<p><small>Narrativa | Wednesday, May 27, 2020 </small></p><p>The number of COVID-19 cases worldwide is now at <strong>5,554,865</strong> Amongst this number, <strong>2,297,943</strong></strong> have already recovered, according to the data <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">provided by Johns Hopkins University</a>. <b>The US</b> tops the list with <strong>1,681,418</strong> recorded cases, followed by <b> Brazil</b>, <b> Russia</b>, and <b>The United Kingdom</b> with <strong>391,222, 370,680, and 266,599</strong> cases respectively.</p><h4>COVID status across most infected countries</h4><p><table width=\"100%\"><thead><tr><th><strong>Active Cases</strong></th><th colspan=\"2\"><strong>Infected</strong></th><th colspan=\"2\"><strong>Deaths</strong></th><th colspan=\"2\"><strong>Recovered</strong></th></tr><tr><th>Total</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th></tr></thead><tbody><tr><td>1. The US: 1,197,587</td><td>1. The US: 1,681,418</td><td>1. Russia: 8,338</td><td>1. The US: 98,929</td><td>1. Russia: 161</td><td>1. The US: 384,902</strong></td><td>1. Russia: 11,079</strong></td></tr><tr><td>2. The United Kingdom: 228,307</td><td>2. Brazil: 391,222</td><td>2. Pakistan: 1,446</td><td>2. The United Kingdom: 37,130</td><td>2. France: 83</td><td>2. Germany: 162,690</strong></td><td>2. Italy: 2,677</strong></td></tr><tr><td>3. Russia: 224,504</td><td>3. Russia: 370,680</td><td>3. The United Kingdom: 1,372</td><td>3. Italy: 32,955</td><td>3. The United Kingdom: 82</td><td>3. Brazil: 158,593</strong></td><td>3. Mexico: 2,256</strong></td></tr></tbody></table></p><p>Moreover, the last 24 hours have seen <strong>19,546</strong> new cases added in the last 24 hours, an increase of <strong>0.35</strong>% from the previous day. <strong> Russia</strong> is the country with the most new cases, having now confirmed <strong>8,338</strong>, followed by <b> Pakistan</b> with <strong>1,446</strong>. <strong>The United Kingdom</strong> and <strong>The US</strong> are not far behind, with <strong>1,117</strong> and <strong>1,117</strong> new patients respectively.</p><p> </p><p>So far, <strong> 341,008 people</strong> have died from the coronavirus worldwide. <strong>The US</strong> is the country with the highest count so far with <strong>98,929</strong> deceased, followed by <strong>The United Kingdom, Italy, and Spain</strong> with <strong>37,130, 32,955, and 27,940</strong> deaths respectively.</p><p>The first news of this new strain of the virus appeared in <strong>China</strong>, in the city of <strong>Wuhan</strong>, on <strong>31 December 2019</strong>. Since that date, the number of infections have multiplied across the continent and the world. At the moment, cases in China have increased to <strong>1</strong> new cases since yesterday. Overall, <strong>105</strong> people are infected with coronavirus.</p><hr><p><strong>Note</strong>: This article has been <strong><u>generated automatically by <a href=\"https://www.narrativa.com\" target=\"_blank\">Narrativa</a></u></strong> from the data by <a href=\"https://www.mscbs.gob.es/profesionales/saludPublica/ccayes/alertasActual/nCov-China/situacionActual.htm\" target=\"_blank\">Ministerio de Sanidad</a>, <a href=\"https://github.com/pcm-dpc/COVID-19\" target=\"_blank\">Dipartimento della Protezione Civile de Italia</a>, <a href=\"https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/Gesamt.html\" target=\"_blank\">Robert Koch Institute</a> and <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">Johns Hopkins University</a>.</p>',\n    'picurl_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.%3A%201%2C681%2C418&top_paises2=Brasil%3A%20391%2C222&top_paises3=Rusia%3A%20370%2C680&client=default&region=world',\n    'picurl': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_comunidades=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&client=default&region=spain',\n    'metatitle': '',\n    'subtitle': '',\n    'picurl_banner_web': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&region_confirmed_num=233%2C037&region_confirmed_today_num=344&region_deaths_today_num=48&region_deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.&top_paises2=Brasil&top_paises3=Rusia&top_paises1_num=1%2C681%2C418&top_paises2_num=391%2C222&top_paises3_num=370%2C680&top_comunidades1=Madrid&top_comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=11%3A03&client=default_web&region=world',\n    '_id': '6658f6eb15b745e282dee2ccf67bb59d',\n    'id': '2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC'}]}}"
     },
     "metadata": {},
     "execution_count": 85
    }
   ],
   "source": [
    "with open('data_indented.json', 'r+') as outfile:\n",
    "    json_readed = json.load(outfile)\n",
    "json_readed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "                                                   data\npost  [{'date': '2020-07-12T13:02:22+02:00', 'metada...",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>data</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>post</th>\n      <td>[{'date': '2020-07-12T13:02:22+02:00', 'metada...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "metadata": {},
     "execution_count": 86
    }
   ],
   "source": [
    "df_json = pd.read_json(\"data_indented.json\")\n",
    "df_json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "                                                   data\npost  [{'date': '2020-07-12T13:02:22+02:00', 'metada...",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>data</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>post</th>\n      <td>[{'date': '2020-07-12T13:02:22+02:00', 'metada...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "metadata": {},
     "execution_count": 87
    }
   ],
   "source": [
    "df_json = pd.read_json(\"data_indented.json\")\n",
    "df_json"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data Mining & Data Wrangling\n",
    "As you can see, there are jsons inside the original json. For that, we have to mining the json (data mining) to be able to use it correctly (data wrangling). \n",
    "\n",
    "How do you solve this issue? Research about this and try a solution. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "                                                   data\npost  [{'date': '2020-07-12T13:02:22+02:00', 'metada...",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>data</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>post</th>\n      <td>[{'date': '2020-07-12T13:02:22+02:00', 'metada...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "metadata": {},
     "execution_count": 88
    }
   ],
   "source": [
    "df = pd.DataFrame(json_readed) \n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "post    [{'date': '2020-07-12T13:02:22+02:00', 'metada...\nName: data, dtype: object"
     },
     "metadata": {},
     "execution_count": 98
    }
   ],
   "source": [
    "df_json[\"data\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "pandas.core.series.Series"
     },
     "metadata": {},
     "execution_count": 95
    }
   ],
   "source": [
    "type(df_json[\"data\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": "post\n"
    }
   ],
   "source": [
    "with open(\"data_indented.json\", \"r\")as outfile:\n",
    "    json_readed = json.load(outfile)\n",
    "    for dict_ in json_readed[\"data\"]:\n",
    "        print(dict_)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "op_paises2_num=391%2C222&top_paises3_num=370%2C680&top_comunidades1=Madrid&top_comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=13%3A10&client=default_web&region=world',\n   '_id': '6658f6eb15b745e282dee2ccf6f0b3c0',\n   'id': '2020-5-27-13_01-COVID-world-en_generic_COVID_EN_GENERIC'},\n  {'date': '2020-05-27T12:41:04+02:00',\n   'metadata': {'picurl_square_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&actives_num=54%2C721&recovered_num=150%2C376&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default',\n    'picurl_square_spain_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default&extended=true',\n    'graphs': {'graph_picurl_china': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china.png',\n     'graph_picurl_us': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us.png',\n     'graph_picurl_deaths_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_evolution.png',\n     'graph_picurl_world': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_world.png',\n     'graph_picurl_china_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_confirmed.png',\n     'graph_picurl_us_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_confirmed.png',\n     'graph_picurl_spain_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_deaths.png',\n     'graph_picurl_us_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_recovered.png',\n     'graph_picurl_deaths_log_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_log_evolution.png',\n     'graph_picurl_italy': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy.png',\n     'graph_picurl_italy_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_recovered.png',\n     'graph_picurl_us_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_us_deaths.png',\n     'graph_picurl_italy_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_deaths.png',\n     'graph_picurl_china_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_deaths.png',\n     'graph_picurl_italy_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_confirmed.png',\n     'graph_picurl_spain_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_confirmed.png',\n     'graph_picurl_china_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC_china_recovered.png'},\n    'picurl_square_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default',\n    'picurl_16_9_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&top_countries=EE.UU.%20%281%2C681%2C418%29%2C%20Brasil%20%28391%2C222%29%2C%20Rusia%20%28370%2C680%29&top_spain=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&region=spain&aspect_ratio=16_9&client=default',\n    'composition_type': 'world',\n    'country_selected': None,\n    'picurl_square_world_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default&extended=true',\n    'state_selected': None},\n   'author': 'Narrativa',\n   '_rev': '1-2f39e9c455969a32a34a2b654f77c1d0',\n   'language': 'en_generic',\n   'title': 'Coronavirus outbreak latest: <strong>5,557,329</strong> people infected worldwide',\n   'content': '<p><small>Narrativa | Wednesday, May 27, 2020 </small></p><p>The number of COVID-19 cases worldwide is now at <strong>5,557,329</strong>, of which <strong>2,298,685</strong></strong> have already recovered, according to the data <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">provided by Johns Hopkins University</a>. <b>The US</b> has the most cases with <strong>1,681,418</strong>, followed by <b> Brazil</b>, <b> Russia</b>, and <b>The United Kingdom</b>, who have confirmed <strong>391,222, 370,680, and 266,599</strong> cases respectively.</p><h4>Status in the world</h4><p><table width=\"100%\"><thead><tr><th><strong>Currently Infected</strong></th><th colspan=\"2\"><strong>Infected</strong></th><th colspan=\"2\"><strong>Deaths</strong></th><th colspan=\"2\"><strong>Recovered</strong></th></tr><tr><th>Total</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th></tr></thead><tbody><tr><td>1. The US: 1,197,587</td><td>1. The US: 1,681,418</td><td>1. Russia: 8,338</td><td>1. The US: 98,929</td><td>1. Russia: 161</td><td>1. The US: 384,902</strong></td><td>1. Russia: 11,079</strong></td></tr><tr><td>2. The United Kingdom: 228,307</td><td>2. Brazil: 391,222</td><td>2. Bangladesh: 1,541</td><td>2. The United Kingdom: 37,130</td><td>2. France: 83</td><td>2. Germany: 162,690</strong></td><td>2. Italy: 2,677</strong></td></tr><tr><td>3. Russia: 224,504</td><td>3. Russia: 370,680</td><td>3. Pakistan: 1,446</td><td>3. Italy: 32,955</td><td>3. The United Kingdom: 82</td><td>3. Brazil: 158,593</strong></td><td>3. Mexico: 2,256</strong></td></tr></tbody></table></p><p>Moreover, <strong>22,010</strong> new positive cases have been added in the last 24 hours, demonstrating a <strong>0.4</strong>% growth from the previous day. <strong> Russia</strong> is the country with the most new cases, having now confirmed <strong>8,338</strong>, followed by <b> Bangladesh</b> with <strong>1,541</strong>. <strong> Pakistan</strong> and <strong>The United Kingdom</strong> are not far behind, with <strong>1,372</strong> and <strong>1,372</strong> new patients respectively.</p><p> </p><p>So far, <strong> 341,118 people</strong> have died from the coronavirus worldwide. <strong>The US</strong> has the most victims so far with <strong>98,929</strong> deceased, followed by <strong>The United Kingdom, Italy, and Spain</strong> with <strong>37,130, 32,955, and 27,940</strong> deaths respectively.</p><p>The first evidences about this new strain of the virus appeared in <strong>China</strong>, in the city of <strong>Wuhan</strong>, on <strong>31 December 2019</strong>. Since then, the infection has spread across Asia and the world. At the moment, cases in China have grown to <strong>1</strong> new cases since Tuesday. Overall, <strong>105</strong> people are infected with the virus.</p><hr><p><strong>Note</strong>: This article has been <strong><u>generated automatically by <a href=\"https://www.narrativa.com\" target=\"_blank\">Narrativa</a></u></strong> from the data by <a href=\"https://www.mscbs.gob.es/profesionales/saludPublica/ccayes/alertasActual/nCov-China/situacionActual.htm\" target=\"_blank\">Ministerio de Sanidad</a>, <a href=\"https://github.com/pcm-dpc/COVID-19\" target=\"_blank\">Dipartimento della Protezione Civile de Italia</a>, <a href=\"https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/Gesamt.html\" target=\"_blank\">Robert Koch Institute</a> and <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">Johns Hopkins University</a>.</p>',\n   'picurl_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.%3A%201%2C681%2C418&top_paises2=Brasil%3A%20391%2C222&top_paises3=Rusia%3A%20370%2C680&client=default&region=world',\n   'picurl': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_comunidades=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&client=default&region=spain',\n   'metatitle': '',\n   'subtitle': '',\n   'picurl_banner_web': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C557%2C329&actives_num=2%2C917%2C526&recovered_num=2%2C298%2C685&deaths_num=341%2C118&region_confirmed_num=233%2C037&region_confirmed_today_num=344&region_deaths_today_num=48&region_deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.&top_paises2=Brasil&top_paises3=Rusia&top_paises1_num=1%2C681%2C418&top_paises2_num=391%2C222&top_paises3_num=370%2C680&top_comunidades1=Madrid&top_comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=12%3A36&client=default_web&region=world',\n   '_id': '6658f6eb15b745e282dee2ccf6def733',\n   'id': '2020-5-27-12_01-COVID-world-en_generic_COVID_EN_GENERIC'},\n  {'date': '2020-05-27T11:06:53+02:00',\n   'metadata': {'picurl_square_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&actives_num=54%2C721&recovered_num=150%2C376&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default',\n    'picurl_square_spain_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&region=spain&aspect_ratio=square&client=default&extended=true',\n    'graphs': {'graph_picurl_china': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china.png',\n     'graph_picurl_us': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us.png',\n     'graph_picurl_deaths_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_evolution.png',\n     'graph_picurl_world': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_world.png',\n     'graph_picurl_china_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_confirmed.png',\n     'graph_picurl_us_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_confirmed.png',\n     'graph_picurl_spain_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_deaths.png',\n     'graph_picurl_us_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_recovered.png',\n     'graph_picurl_deaths_log_evolution': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_deaths_log_evolution.png',\n     'graph_picurl_italy': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy.png',\n     'graph_picurl_italy_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_recovered.png',\n     'graph_picurl_us_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_us_deaths.png',\n     'graph_picurl_italy_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_deaths.png',\n     'graph_picurl_china_deaths': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_deaths.png',\n     'graph_picurl_italy_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_italy_confirmed.png',\n     'graph_picurl_spain_confirmed': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_spain_confirmed.png',\n     'graph_picurl_china_recovered': 'https://resources.gabriele.ai/coronavirus/2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC_china_recovered.png'},\n    'picurl_square_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default',\n    'picurl_16_9_spain': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&top_countries=EE.UU.%20%281%2C681%2C418%29%2C%20Brasil%20%28391%2C222%29%2C%20Rusia%20%28370%2C680%29&top_spain=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&region=spain&aspect_ratio=16_9&client=default',\n    'composition_type': 'world',\n    'country_selected': None,\n    'picurl_square_world_extended': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&region=world&aspect_ratio=square&client=default&extended=true',\n    'state_selected': None},\n   'author': 'Narrativa',\n   '_rev': '1-e9ca35acd2d130b04289e3333083ca14',\n   'language': 'en_generic',\n   'title': 'Coronavirus outbreak latest: <strong>341,008</strong> deaths confirmed around the world',\n   'content': '<p><small>Narrativa | Wednesday, May 27, 2020 </small></p><p>The number of COVID-19 cases worldwide is now at <strong>5,554,865</strong> Amongst this number, <strong>2,297,943</strong></strong> have already recovered, according to the data <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">provided by Johns Hopkins University</a>. <b>The US</b> tops the list with <strong>1,681,418</strong> recorded cases, followed by <b> Brazil</b>, <b> Russia</b>, and <b>The United Kingdom</b> with <strong>391,222, 370,680, and 266,599</strong> cases respectively.</p><h4>COVID status across most infected countries</h4><p><table width=\"100%\"><thead><tr><th><strong>Active Cases</strong></th><th colspan=\"2\"><strong>Infected</strong></th><th colspan=\"2\"><strong>Deaths</strong></th><th colspan=\"2\"><strong>Recovered</strong></th></tr><tr><th>Total</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th><th>Total</th><th>Last 24h</th></tr></thead><tbody><tr><td>1. The US: 1,197,587</td><td>1. The US: 1,681,418</td><td>1. Russia: 8,338</td><td>1. The US: 98,929</td><td>1. Russia: 161</td><td>1. The US: 384,902</strong></td><td>1. Russia: 11,079</strong></td></tr><tr><td>2. The United Kingdom: 228,307</td><td>2. Brazil: 391,222</td><td>2. Pakistan: 1,446</td><td>2. The United Kingdom: 37,130</td><td>2. France: 83</td><td>2. Germany: 162,690</strong></td><td>2. Italy: 2,677</strong></td></tr><tr><td>3. Russia: 224,504</td><td>3. Russia: 370,680</td><td>3. The United Kingdom: 1,372</td><td>3. Italy: 32,955</td><td>3. The United Kingdom: 82</td><td>3. Brazil: 158,593</strong></td><td>3. Mexico: 2,256</strong></td></tr></tbody></table></p><p>Moreover, the last 24 hours have seen <strong>19,546</strong> new cases added in the last 24 hours, an increase of <strong>0.35</strong>% from the previous day. <strong> Russia</strong> is the country with the most new cases, having now confirmed <strong>8,338</strong>, followed by <b> Pakistan</b> with <strong>1,446</strong>. <strong>The United Kingdom</strong> and <strong>The US</strong> are not far behind, with <strong>1,117</strong> and <strong>1,117</strong> new patients respectively.</p><p> </p><p>So far, <strong> 341,008 people</strong> have died from the coronavirus worldwide. <strong>The US</strong> is the country with the highest count so far with <strong>98,929</strong> deceased, followed by <strong>The United Kingdom, Italy, and Spain</strong> with <strong>37,130, 32,955, and 27,940</strong> deaths respectively.</p><p>The first news of this new strain of the virus appeared in <strong>China</strong>, in the city of <strong>Wuhan</strong>, on <strong>31 December 2019</strong>. Since that date, the number of infections have multiplied across the continent and the world. At the moment, cases in China have increased to <strong>1</strong> new cases since yesterday. Overall, <strong>105</strong> people are infected with coronavirus.</p><hr><p><strong>Note</strong>: This article has been <strong><u>generated automatically by <a href=\"https://www.narrativa.com\" target=\"_blank\">Narrativa</a></u></strong> from the data by <a href=\"https://www.mscbs.gob.es/profesionales/saludPublica/ccayes/alertasActual/nCov-China/situacionActual.htm\" target=\"_blank\">Ministerio de Sanidad</a>, <a href=\"https://github.com/pcm-dpc/COVID-19\" target=\"_blank\">Dipartimento della Protezione Civile de Italia</a>, <a href=\"https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/Gesamt.html\" target=\"_blank\">Robert Koch Institute</a> and <a href=\"https://systems.jhu.edu/research/public-health/ncov/\" target=\"_blank\">Johns Hopkins University</a>.</p>',\n   'picurl_world': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.%3A%201%2C681%2C418&top_paises2=Brasil%3A%20391%2C222&top_paises3=Rusia%3A%20370%2C680&client=default&region=world',\n   'picurl': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=233%2C037&confirmed_today_num=344&deaths_today_num=48&deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_comunidades=Madrid%20%2867%2C049%29%2C%20Catalu%C3%B1a%20%2855%2C888%29%2C%20Castilla%20y%20Le%C3%B3n%20%2818%2C627%29&client=default&region=spain',\n   'metatitle': '',\n   'subtitle': '',\n   'picurl_banner_web': 'https://4i2g925ohj.execute-api.eu-west-3.amazonaws.com/pro/coronavirus?confirmed_num=5%2C554%2C865&actives_num=2%2C915%2C914&recovered_num=2%2C297%2C943&deaths_num=341%2C008&region_confirmed_num=233%2C037&region_confirmed_today_num=344&region_deaths_today_num=48&region_deaths_num=27%2C940&date=27%20de%20mayo%20del%202020&top_paises1=EE.UU.&top_paises2=Brasil&top_paises3=Rusia&top_paises1_num=1%2C681%2C418&top_paises2_num=391%2C222&top_paises3_num=370%2C680&top_comunidades1=Madrid&top_comunidades2=Catalu%C3%B1a&top_comunidades3=Castilla%20y%20Le%C3%B3n&top_comunidades1_num=67%2C049&top_comunidades2_num=55%2C888&top_comunidades3_num=18%2C627&gen_date=2020/05/27&gen_time=11%3A03&client=default_web&region=world',\n   '_id': '6658f6eb15b745e282dee2ccf67bb59d',\n   'id': '2020-5-27-11_01-COVID-world-en_generic_COVID_EN_GENERIC'}]}"
     },
     "metadata": {},
     "execution_count": 99
    }
   ],
   "source": [
    "with open(\"data_indented.json\", \"r+\")as outfile:\n",
    "    json_readed=json.load(outfile)\n",
    "    a_json_ = json_readed[\"data\"]\n",
    "a_json_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"final_data.json\", \"w+\") as outfile:\n",
    "    json.dump(a_json_, outfile, indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "                                                  post\n0    {'date': '2020-07-12T13:02:22+02:00', 'metadat...\n1    {'date': '2020-07-12T12:02:30+02:00', 'metadat...\n2    {'date': '2020-07-12T11:11:18+02:00', 'metadat...\n3    {'date': '2020-07-12T10:17:09+02:00', 'metadat...\n4    {'date': '2020-07-12T09:17:23+02:00', 'metadat...\n..                                                 ...\n995  {'date': '2020-05-27T15:25:16+02:00', 'metadat...\n996  {'date': '2020-05-27T14:18:34+02:00', 'metadat...\n997  {'date': '2020-05-27T13:13:22+02:00', 'metadat...\n998  {'date': '2020-05-27T12:41:04+02:00', 'metadat...\n999  {'date': '2020-05-27T11:06:53+02:00', 'metadat...\n\n[1000 rows x 1 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>post</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>{'date': '2020-07-12T13:02:22+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>{'date': '2020-07-12T12:02:30+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>{'date': '2020-07-12T11:11:18+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>{'date': '2020-07-12T10:17:09+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>{'date': '2020-07-12T09:17:23+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>995</th>\n      <td>{'date': '2020-05-27T15:25:16+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>996</th>\n      <td>{'date': '2020-05-27T14:18:34+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>997</th>\n      <td>{'date': '2020-05-27T13:13:22+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>998</th>\n      <td>{'date': '2020-05-27T12:41:04+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>999</th>\n      <td>{'date': '2020-05-27T11:06:53+02:00', 'metadat...</td>\n    </tr>\n  </tbody>\n</table>\n<p>1000 rows × 1 columns</p>\n</div>"
     },
     "metadata": {},
     "execution_count": 106
    }
   ],
   "source": [
    "df_a_json_=pd.read_json(\"final_data.json\")\n",
    "df_a_json_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "                                                   data  \\\npost  [{'date': '2020-07-12T13:02:22+02:00', 'metada...   \n0                                                   NaN   \n1                                                   NaN   \n2                                                   NaN   \n3                                                   NaN   \n...                                                 ...   \n995                                                 NaN   \n996                                                 NaN   \n997                                                 NaN   \n998                                                 NaN   \n999                                                 NaN   \n\n                                                   post  \npost                                                NaN  \n0     {'date': '2020-07-12T13:02:22+02:00', 'metadat...  \n1     {'date': '2020-07-12T12:02:30+02:00', 'metadat...  \n2     {'date': '2020-07-12T11:11:18+02:00', 'metadat...  \n3     {'date': '2020-07-12T10:17:09+02:00', 'metadat...  \n...                                                 ...  \n995   {'date': '2020-05-27T15:25:16+02:00', 'metadat...  \n996   {'date': '2020-05-27T14:18:34+02:00', 'metadat...  \n997   {'date': '2020-05-27T13:13:22+02:00', 'metadat...  \n998   {'date': '2020-05-27T12:41:04+02:00', 'metadat...  \n999   {'date': '2020-05-27T11:06:53+02:00', 'metadat...  \n\n[1001 rows x 2 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>data</th>\n      <th>post</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>post</th>\n      <td>[{'date': '2020-07-12T13:02:22+02:00', 'metada...</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>0</th>\n      <td>NaN</td>\n      <td>{'date': '2020-07-12T13:02:22+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NaN</td>\n      <td>{'date': '2020-07-12T12:02:30+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NaN</td>\n      <td>{'date': '2020-07-12T11:11:18+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NaN</td>\n      <td>{'date': '2020-07-12T10:17:09+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>995</th>\n      <td>NaN</td>\n      <td>{'date': '2020-05-27T15:25:16+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>996</th>\n      <td>NaN</td>\n      <td>{'date': '2020-05-27T14:18:34+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>997</th>\n      <td>NaN</td>\n      <td>{'date': '2020-05-27T13:13:22+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>998</th>\n      <td>NaN</td>\n      <td>{'date': '2020-05-27T12:41:04+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>999</th>\n      <td>NaN</td>\n      <td>{'date': '2020-05-27T11:06:53+02:00', 'metadat...</td>\n    </tr>\n  </tbody>\n</table>\n<p>1001 rows × 2 columns</p>\n</div>"
     },
     "metadata": {},
     "execution_count": 116
    }
   ],
   "source": [
    "final_df=pd.concat([df_json, df_a_json_], axis=1, sort=False)\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "                                                   post\npost                                                NaN\n0     {'date': '2020-07-12T13:02:22+02:00', 'metadat...\n1     {'date': '2020-07-12T12:02:30+02:00', 'metadat...\n2     {'date': '2020-07-12T11:11:18+02:00', 'metadat...\n3     {'date': '2020-07-12T10:17:09+02:00', 'metadat...\n...                                                 ...\n995   {'date': '2020-05-27T15:25:16+02:00', 'metadat...\n996   {'date': '2020-05-27T14:18:34+02:00', 'metadat...\n997   {'date': '2020-05-27T13:13:22+02:00', 'metadat...\n998   {'date': '2020-05-27T12:41:04+02:00', 'metadat...\n999   {'date': '2020-05-27T11:06:53+02:00', 'metadat...\n\n[1001 rows x 1 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>post</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>post</th>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>0</th>\n      <td>{'date': '2020-07-12T13:02:22+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>{'date': '2020-07-12T12:02:30+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>{'date': '2020-07-12T11:11:18+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>{'date': '2020-07-12T10:17:09+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>995</th>\n      <td>{'date': '2020-05-27T15:25:16+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>996</th>\n      <td>{'date': '2020-05-27T14:18:34+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>997</th>\n      <td>{'date': '2020-05-27T13:13:22+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>998</th>\n      <td>{'date': '2020-05-27T12:41:04+02:00', 'metadat...</td>\n    </tr>\n    <tr>\n      <th>999</th>\n      <td>{'date': '2020-05-27T11:06:53+02:00', 'metadat...</td>\n    </tr>\n  </tbody>\n</table>\n<p>1001 rows × 1 columns</p>\n</div>"
     },
     "metadata": {},
     "execution_count": 117
    }
   ],
   "source": [
    "final_df=final_df.drop([\"data\"],axis=1)\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python",
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "version": "3.8.3-final"
  },
  "orig_nbformat": 2,
  "file_extension": ".py",
  "mimetype": "text/x-python",
  "name": "python",
  "npconvert_exporter": "python",
  "pygments_lexer": "ipython3",
  "version": 3,
  "kernelspec": {
   "name": "python38364bitcbfd2998b9c747eaa6ef61b7425e754e",
   "display_name": "Python 3.8.3 64-bit"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}